src/climate_learn/loaders.py

```diff
@@ -0,0 +1,356 @@
+    data_module: Union[DataModule, IterDataModule],
+    preset: Optional[str] = None,
+    model: Optional[Union[str, nn.Module]] = None,
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    optim: Optional[Union[str, torch.optim.Optimizer]] = None,
+    optim_kwargs: Optional[Dict[str, Any]] = None,
+    sched: Optional[Union[str, LRScheduler._LRScheduler]] = None,
+    sched_kwargs: Optional[Dict[str, Any]] = None,
+    train_loss: Optional[Union[str, Callable]] = None,
+    val_loss: Optional[Union[Iterable[str], Iterable[Callable]]] = None,
+    test_loss: Optional[Union[Iterable[str], Iterable[Callable]]] = None,
+    train_target_transform: Optional[Union[str, Callable]] = None,
+    val_target_transform: Optional[
+        Union[Iterable[str], Iterable[Callable]]
+    ] = None,
+    test_target_transform: Optional[
+        Union[Iterable[str], Iterable[Callable]]
+    ] = None
+):
+    # Load the model
+    if preset is None and model is None:
+        raise RuntimeError("Please specify 'preset' or 'model'")
+    elif preset:
+        print(f"Loading preset: {preset}")
+        model, optimizer, lr_scheduler = load_preset(task, data_module, preset)
+    elif isinstance(model, str):
+        print(f"Loading model: {model}")
+        model_cls = MODEL_REGISTRY.get(model, None)
+        if model_cls is None:
+            raise NotImplementedError(
+                f"{model} is not an implemented model. If you think it should be,"
+                " please raise an issue at"
+                " https://github.com/aditya-grover/climate-learn/issues."
+            )
+        model = model_cls(**model_kwargs)
+    elif isinstance(model, nn.Module):
+        print("Using custom network")
+    else:
+        raise TypeError("'model' must be str or nn.Module")
+    # Load the optimizer
+    if preset is None and optim is None:
+        raise RuntimeError("Please specify 'preset' or 'optim'")
+    elif preset:
+        print("Using preset optimizer")
+    elif isinstance(optim, str):
+        print(f"Loading optimizer {optim}")
+        optimizer = load_optimizer(model, optim, optim_kwargs)
+    elif isinstance(optim, torch.optim.Optimizer):
+        print("Using custom optimizer")
+    else:
+        raise TypeError("'optim' must be str or torch.optim.Optimizer")
+    # Load the LR scheduler
+    if preset is None and sched is None:
+        raise RuntimeError("Please specify 'preset' or 'sched'")
+    elif preset:
+            print("Using preset learning rate scheduler")            
+    elif isinstance(sched, str):
+        print(f"Loading learning rate scheduler: {sched}")
+        lr_scheduler = load_lr_scheduler(sched, optimizer, sched_kwargs)
+    elif isinstance(sched, LRScheduler._LRScheduler):
+        print("Using custom learning rate scheduler")
+    else:
+        raise TypeError("'sched' must be str or torch.optim.lr_scheduler._LRScheduler")
+    # Load training loss
+    in_vars, out_vars = get_data_variables(data_module)
```

Comment:
How should I inspect whether the datasets have been setup? I don't see a method in `DataModule` that would make this easy.

Reply:
Well for now, you can call the `get_lat_lon()` as it would return `None`. Once, this PR is merged, I would open a new PR which would then add a method corresponding to this check. I would also then change the logic from `get_lat_lon()` to that new method.